app/upgrades/v15/upgrades.go

```diff
@@ -21,7 +24,33 @@ func CreateUpgradeHandler(
 			return vm, err
 		}
 
-		ctx.Logger().Info("Upgrade complete")
+		UpgradeMinCommissionRate(ctx, *keepers.StakingKeeper)
+
+		ctx.Logger().Info("Upgrade v15 complete")
 		return vm, err
 	}
 }
+
+// UpgradeMinCommissionRate sets the minimum commission rate staking parameter to 5%
+// and updates the commission rate for all validators that have a commission rate less than 5%
+func UpgradeMinCommissionRate(ctx sdk.Context, sk stakingkeeper.Keeper) {
+	params := sk.GetParams(ctx)
```

Comment:
Add log messages at the beginning and end of the `UpgradeMinCommissionRate` method, e.g., 
```go
unc UpgradeMinCommissionRate(ctx sdk.Context, sk stakingkeeper.Keeper) {
	ctx.Logger().Info("Migrating min commission rate...")
	// ... 
	ctx.Logger().Info("Finished migrating min commission rate")
```
If it fails, we need to identify fast the bug.